IPython/terminal/embed.py

```diff
@@ -77,44 +124,56 @@ class InteractiveShellEmbed(TerminalInteractiveShell):
 
     @property
     def embedded_active(self):
-        return self._call_location_id not in InteractiveShellEmbed._inactive_locations
+        return (self._call_location_id not in InteractiveShellEmbed._inactive_locations)\
+            and (self._init_location_id not in InteractiveShellEmbed._inactive_locations)
+
+    def _disable_init_location(self):
+        """Disable the current Instance creation location"""
+        InteractiveShellEmbed._inactive_locations.add(self._init_location_id)
 
     @embedded_active.setter
     def embedded_active(self, value):
-        if value :
+        if value:
             if self._call_location_id in InteractiveShellEmbed._inactive_locations:
-                InteractiveShellEmbed._inactive_locations.remove(self._call_location_id)
+                InteractiveShellEmbed._inactive_locations.remove(
+                    self._call_location_id)
```

Comment:
BTW, sets have a `.discard()` method which is like `.remove()` but doesn't raise if the item isn't in the set. So instead of checking `if a in b: b.remove(a)`, we can just do `b.discard(a)`.